packages/app/universal-ts-utils/src/array/sortByField.ts

```diff
@@ -0,0 +1,49 @@
+type KeysMatching<T extends object, V> = {
+  [K in keyof T]: T[K] extends V ? K : never
+}[keyof T]
+
+/*
+ TODO: Extend this method more types of fields.
+ For now, it only supports fields of type string as it is the most common use case.
+ the objective is to use the compare method defined on `sort.ts` but don't want to expose it at this point.
+
+ So in a follow-up will try to come up with a solution to have "internal" not exposed utilities, and then extends this
+ method.
+*/
```

Comment:
Will come up with a proposal for this in a separate PR

Reply:
After all discussions looks like this is not going to be needed for now, so postponing it